docs/dev/clusteroperator.md

```diff
@@ -97,3 +97,22 @@ If an error blocks reaching 4.0.1, the conditions might be:
 * `Progressing` is true with message `Unable to apply 4.0.1: a required object is missing`
 
 The progressing message is the first message a human will see when debugging an issue, so it should be terse, succinct, and summarize the problem well.  The failing message can be more verbose. Start with simple, easy to understand messages and grow them over time to capture more detail.
+
+
+#### Conditions and Install/Upgrade
+
+Conditions determine when the CVO considers certain actions complete, the following table summarizes what it looks at and when.
+
+
+| operation | version | available | degraded | progressing | 
+|-----------|---------|-----------|----------|-------------|
+| Install completion[1] | current(whatever was being installed) | true | any | any
+| Begin upgrade | any | any | any | any
+| Begin upgrade (w/ force) | any | any | any | any 
+| Upgrade completion[2]| newVersion(target version for the upgrade) | true | false | false
+
+[1] Install cannot proceed to cluster operators in the next runlevel until all cluster operators in the current runlevel achieve these conditions 
```

Comment:
@abhinavdahiya updated ^^

Reply:
I'm confused what you mean here - ordering only matters during upgrade, it's ignored on install and sync?  Maybe I'm misinterpreting what you mean by run level

Reply:
> ordering only matters during upgrade, it's ignored on install and sync

that's what i was trying to say, yes.  By runlevel i meant like "70_blah".  My understanding was everything at run level 70 runs in parallel during install.  then everything at runlevel 71.  etc.

and then during upgrade I was trying to say it's fully ordered, no parallelization.

I wasn't trying to say anything about sync.